smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/inject/LegacyEmitterInjectionTest.java

```diff
@@ -0,0 +1,442 @@
+            emitter.send("c");
+            emitter.complete();
+        }
+
+        @Incoming("foo")
+        public void consume(final String s) {
+            list.add(s);
+        }
+    }
+
+    @ApplicationScoped
+    public static class MyBeanEmittingMessages {
+        @Inject
+        @Channel("foo")
+        Emitter<Message<String>> emitter;
+        private final List<String> list = new CopyOnWriteArrayList<>();
+
+        public Emitter<Message<String>> emitter() {
+            return emitter;
+        }
+
+        public List<String> list() {
+            return list;
+        }
+
+        public void run() {
+            emitter.send(Message.of("a"));
+            emitter.send(Message.of("b"));
+            emitter.send(Message.of("c"));
+
+        }
+
+        @Incoming("foo")
+        public void consume(final String s) {
+            list.add(s);
+        }
+    }
+
+    @ApplicationScoped
+    public static class MyBeanEmittingMessagesUsingStream {
+        @Inject
+        @Channel("foo")
+        Emitter<Message<String>> emitter;
+        private final List<String> list = new CopyOnWriteArrayList<>();
+
+        public Emitter<Message<String>> emitter() {
+            return emitter;
+        }
+
+        public List<String> list() {
+            return list;
+        }
+
+        public void run() {
+            emitter.send(Message.of("a"));
+            emitter.send(Message.of("b"));
+            emitter.send(Message.of("c"));
+        }
+
+        @Incoming("foo")
+        public void consume(final String s) {
+            list.add(s);
+        }
+    }
+
+    public static class BeanWithMissingStream {
+        @Inject
+        @Stream("missing")
+        Emitter<Message<String>> emitter;
+        private final List<String> list = new CopyOnWriteArrayList<>();
+
+        public Emitter<Message<String>> emitter() {
+            return emitter;
+        }
+    }
+
+    public static class BeanWithMissingChannel {
+        @Inject
+        @Channel("missing")
+        Emitter<Message<String>> emitter;
+        private final List<String> list = new CopyOnWriteArrayList<>();
+
+        public Emitter<Message<String>> emitter() {
+            return emitter;
+        }
+    }
+
+    @ApplicationScoped
+    public static class MyBeanEmittingNull {
+        @Inject
+        @Channel("foo")
+        Emitter<String> emitter;
+        private final List<String> list = new CopyOnWriteArrayList<>();
+        private boolean caught;
+
+        public Emitter<String> emitter() {
+            return emitter;
+        }
+
+        public boolean isCaught() {
+            return true;
+        }
+
+        public List<String> list() {
+            return list;
+        }
+
+        public void run() {
+            emitter.send("a");
+            emitter.send("b");
+            /*
+             * try { emitter.send(null); } catch (IllegalArgumentException e) { caught =
+             * true; }
+             */
```

Comment:
this is intentionally commented? On line https://github.com/smallrye/smallrye-reactive-messaging/pull/339/files#diff-4a279cdc72ce161238860cec0e680dcdR108 it seems to be checked whether caught == true.

Reply:
Yes, it is. With the new change, you cannot pass 'null' any more, because compiler cannot figure out which method to match null because of the method overloading.
@cescoffier maybe it is better to delete this block. My bad for leaving them.

Reply:
Great catch, fixing...

Reply:
Fixed.